Second Challenge (#4)
Joshua-Medvinsky/Forta-Bots-Challenges: README.md

## Description

> This agent detects when a swap is made on Uniswap V3 pools

Reviewer: ```suggestion
This bot detects when a swap is made on Uniswap V3 pools
```

## Supported Chains

- Ethereum

> - Polygon

Reviewer: if this bot will work in Polygon, then its chainId must be included in `package.json`

## Alerts

> Describe each of the type of alerts fired by this agent

Reviewer: ```suggestion
```

- FORTA-1

>   - Fired when a new swap is made onUniswap V3 pools

Reviewer: ```suggestion
  - Fired when a new swap is made on Uniswap V3 pools
```

  - Severity is always set to "low" 
  - Type is always set to "info" 
  - Metadata:
    -sender: The initiator of the smart contract call.
    -recipient: The party receiving the swapped tokens.
    -amount0: The quantity of tokens being exchanged.
    -amount1: The quantity of tokens acquired.
    -poolAddress: The address of the smart contract representing the Uniswap V3 pool.

>     -liquidity: The total liquidity amount involved in the swap.

Reviewer: ```suggestion
    - sender: The initiator of the smart contract call.
    - recipient: The party receiving the swapped tokens.
    - amount0: The quantity of tokens being exchanged.
    - amount1: The quantity of tokens acquired.
    - poolAddress: The address of the smart contract representing the Uniswap V3 pool.
    - liquidity: The total liquidity amount involved in the swap.
```
for markdown to work properly

## Test Data

> The agent behaviour can be verified with the tests in agent.ts.spec

Reviewer: there should be a real transaction hash here. it should work properly with a real Uniswap swap. `npm run tx 0x...`
